Libraries/CommonFunctions.psm1

```diff
@@ -1952,61 +1952,72 @@ function Enable-RootUser {
     return $deploymentResult
 }
 
-function IsGreaterKernelVersion() {
+function Is-DpdkCompatible() {
     param (
-        [string] $actualKernelVersion,
-        [string] $detectedDistro
+        [string] $KernelVersion,
+        [string] $DetectedDistro,
+        [string] $CompatibleDistro
     )
 
-    # Supported Distro and kernel version for DPDK on Azure
-    # https://docs.microsoft.com/en-us/azure/virtual-network/setup-dpdk
-    $SUPPORTED_DISTRO_KERNEL = @{
-        "UBUNTU" = "4.15.0-1015-azure";
-        "SLES" = "4.12.14-5.5-azure";
-        "SLES 15" = "4.12.14-5.5-azure";
-        "SUSE" = "4.12.14-5.5-azure";
-        "REDHAT" = "3.10.0-862.9.1.el7";
-        "CENTOS" = "3.10.0-862.3.3.el7";
-    }
-
-    if ($SUPPORTED_DISTRO_KERNEL.Keys -contains $detectedDistro) {
-        $supportKernelVersions = $SUPPORTED_DISTRO_KERNEL[$detectedDistro] -split "[\.\-]+"
-        $actualKernelVersions = $actualKernelVersion -split "[\.\-]+"
-        for($i=0; $i -lt $supportKernelVersions.Length;$i++) {
-            try {
-                    $supportKernelVersions[$i] = [int]$supportKernelVersions[$i]
-                } catch {
-                    $supportKernelVersions[$i] = 0
-                    continue
-                }
+    if ($CompatibleDistro) {
+        if ($CompatibleDistro.Contains($DetectedDistro)) {
+            Write-LogInfo "Confirmed supported distro: $DetectedDistro"
+            return $true
+        } else {
+            Write-LogWarn "Unsupported distro: $DetectedDistro"
+            return $false
         }
-        for($i=0; $i -lt $actualKernelVersions.Length;$i++) {
-            try {
-                    $actualKernelVersions[$i] = [int]$actualKernelVersions[$i]
-                } catch {
-                    $actualKernelVersions[$i] = 0
-                    continue
-                }
+    } else {
+        # Supported Distro and kernel version for DPDK on Azure
+        # https://docs.microsoft.com/en-us/azure/virtual-network/setup-dpdk
+        $SUPPORTED_DISTRO_KERNEL = @{
```

Comment:
I didn't change below logic, just reorg the code, rename function name.
- For supported distro and kernel version greater and equal given kernel version, then case continue.
- For VERIFY-DPDK-OVS and VERIFY-DPDK-NFF-GO, it only run when distro is ubuntu.
I change code for non supported distro, case will be marked skip.

Reply:
@juhlee-microsoft the reason those values exist in a static way is that they are defined in a static way here: https://docs.microsoft.com/en-us/azure/virtual-network/setup-dpdk . The DPDK matrix is way more complicated and changes daily, but this is not the reason for this patch. This patch addresses a very urgent concern of the tests failing for clear OS or other distros that are not defined at the above link. If you consider that further discussion is needed, please open an issue so that we can address it in the future. This PR is a refactor + skipping the test cases on unsupported platforms like ClearOS.

Reply:
First, you need to make sure if those are only kernel version supporting DPDK, or those are the lowest version among supporting kernels. Ok to close this thread today but we should revisit in future.

Reply:
@LiliDeng @ader1990 are you answering of my question? Please re-read.

Reply:
> @LiliDeng @ader1990 are you answering of my question? Please re-read.

I didn't change the design for the static way check DPDK, so we can file a task to track your concern.

Do you know who posts this document https://docs.microsoft.com/en-us/azure/virtual-network/setup-dpdk? If we don't trust it, we can go through different distro and kernel version, to make sure they are the lowest version which supports DPDK.

If we don't use this static way in PS side, then we will still similar way is bash side. Since the truth is that DPDK not supported in each distro.

Reply:
@LiliDeng the azure dpdk docs page contains generic instructions and minimum kernel versions. The DPDK support matrix is quite complicated, depending on what you want to achieve. For example, primary secondary DPDK processes are supported only from DPDK version 19.05. Some DPDK features are supported only from kernels > 4.20. The minimum supported version of DPDK on Azure should be something like 18.11, as the older ones are unusable/unstable in some scenarios. We should continue this discussion on an email thread with the team.

Reply:
@ader1990 Thanks for clear explanation, I will close this PR then.

Reply:
From my perspective, this PR is correct, as its purpose is to skip the tests on the OSses that it should not run on.

Reply:
@LiliDeng @ader1990 It sounds better now. No need to close this PR, but we should add a new issue how this automation support kernel versions in the run-time. As @ader1990 mentioned, those kernel versions in setup-dpdk doc is min-support one. We will be able to modify the script to support future values.